### DISCUSSION TITLE
Manual Action

### DISCUSSION THREAD

**Discussion by dohnto** [woodpecker-ci/woodpecker#3638] [Ask a Question]
Hi,

first let me thank you for this wonderful piece of software.

I am evaluating Woodpecker and I am interested if there is already a feature to gate some steps with manual action similar to what Gitlab and other offers (https://about.gitlab.com/blog/2021/05/20/dag-manual-fix/) or there is an existing feature request.

Alternatively is it possible to achieve this functionality using plugins?

Thanks for guidance

**Comment by qwerty287** [comment]
No, that's not supported yet in the way gitlab supports it.
What you can do:
- Exec pipelines manually. As you can define additional variables you can get a fine-grained system about what should run and what not. If the step that should require a manual action does not need access to files of the other workflows, this option could work for you.
- Add a special step that blocks the pipeline till your manual action. My idea would be something like this: Run a http server, expose the ports (depending on which backend you use and what your security requirements are you could do that with direct port exposing or with an additional tool like https://tunnelmole.com/) and if you receive a request to that http server exit the step.

I can also convert this discussion into a feature request?

**Reply by zc-devs** [reply]
2. Seems, it requires tuning up [the timeout](https://woodpecker-ci.org/docs/2.3/usage/project-settings#timeout)

**Comment by anbraten** [comment]
Question might also be what you are trying to achieve exactly. The manual and deployment events could be of interest. Maybe the repo protection so approval for pipelines is required might also be a solution.

**Reply by dohnto** [reply]
We are building a pipeline which would deploy an application into several environments (e. g. first run tests, build container image, then some manual action is required and after that deploy to staging environment, another manual action and then deploy to prod1, prod2, prod3, prod4, ... prodN at will). I could leverage the "Deploy" feature of woodpecker but IIUC it will create a new pipeline which is difficult to match to the original one and given we want to deploy to several environments, it is not convenient to manually choose them (type it into the input box of Deploy window).

Also I would like to have a single view on the pipeline deploying to different environments.

I don't see a way to achieve this in woodpecker now out of the box. The custom http server which @qwerty287 suggests might be a solution.

**Comment by qwerty287** [comment]
@dohnto Should I convert this into a feature request or close this?

**Comment by dohnto** [comment]
Hi, unfortunately we decided not to progress with WoodpeckerCI. I am closing this. Thank you.

**Comment by echowings** [comment]
This is a key feature  I wanted it most. however I turn to concourse ci instead of woodpecker
